presto-parquet/src/main/java/com/facebook/presto/parquet/writer/valuewriter/UuidValuesWriter.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.parquet.writer.valuewriter;
+
+import com.facebook.presto.common.block.Block;
+import org.apache.parquet.column.values.ValuesWriter;
+import org.apache.parquet.io.api.Binary;
+import org.apache.parquet.schema.PrimitiveType;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+
+import static io.airlift.slice.SizeOf.SIZE_OF_LONG;
+
+@NotThreadSafe
+public class UuidValuesWriter
+        extends PrimitiveValueWriter
+{
+    private final ByteBuffer writeBuffer = ByteBuffer.allocate(2 * SIZE_OF_LONG)
+            .order(ByteOrder.BIG_ENDIAN);
+
+    public UuidValuesWriter(ValuesWriter valuesWriter, PrimitiveType parquetType)
+    {
+        super(parquetType, valuesWriter);
```

Comment:
Actually, I think the `BooleanValueWriter` is wrong because the superclass `PrimitivateValueWriter` already stores the reference to the `valuesWriter` which is accessed through `getValueWriter()`. Storing it again seems to just be a redundant reference.

https://github.com/prestodb/presto/blob/152e64dbebf2d8c9deedaae833d4ebfcaa4a531f/presto-parquet/src/main/java/com/facebook/presto/parquet/writer/valuewriter/PrimitiveValueWriter.java#L33-L43

In the UUID writer implementation, we just use `getValueWriter()`